### PR TITLE
Feature/tests refactoring

### PR DESCRIPTION
**Pull request by lebdron**
## What is this pull request?
Moves mocks for modules to corresponding folders and other test refactoring
   
## Why do you implement it? Why do we need this pull request?
Removes code duplication, cleaner test code which only shows what is tested

## Details/Features
- TestSubscriber is used in more tests
- More module tests have fixtures

### CODE REVIEW

File: test/module/irohad/torii/torii_service_test.cpp [hunk 1]
```diff
@@ -29,7 +31,6 @@ limitations under the License.
 #include <queries.pb.h>
 
 #include "torii/processor/transaction_processor_impl.hpp"
-#include "mock_classes.hpp"
 
 constexpr const char *Ip = "0.0.0.0";
```
Comment (muratovv) [inline test/module/irohad/torii/torii_service_test.cpp]: Replace with string.

File: test/module/irohad/torii/torii_service_test.cpp [hunk 2]
```diff
@@ -43,15 +44,18 @@ using ::testing::A;
...
 
+using namespace iroha::network;
+using namespace iroha::validation;
+using namespace iroha::ametsuchi;
 
 class ToriiServiceTest : public testing::Test {
  public:
   virtual void SetUp() {
     runner = new ServerRunner(Ip, Port);
     th = std::thread([runner = runner] {
```
Comment (muratovv) [inline test/module/irohad/torii/torii_service_test.cpp]: Replace with `this` closure.

File: test/module/irohad/consensus/yac/yac_mocks.hpp [hunk 3]
```diff
@@ -18,198 +18,189 @@
...
-class FakeTimer : public Timer {
- public:
-  void invokeAfterDelay(uint64_t millis,
-                        std::function<void()> handler) override {
-    handler();
-  };
+        MOCK_METHOD0(deny, void());
 
-  MOCK_METHOD0(deny, void());
+        MockTimer() {}
```
Comment (lebdron) [inline test/module/irohad/consensus/yac/yac_mocks.hpp]: We can review all constructors and assignment operators of mocks to leave only required ones, and add comments why they are required

File: test/module/irohad/model/command_validate_execute_test.cpp [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+
+  decltype(AccountAsset().balance) balance = 150ul;
+  Asset asset;
+  AccountAsset wallet;
+
+  std::shared_ptr<AddAssetQuantity> add_asset_quantity;
+};
+
+TEST_F(AddAssetQuantityTest, ValidWhenNewWallet) {
+  // Case 1. Add asset first time - no wallet
```
Comment (muratovv) [inline test/module/irohad/model/command_validate_execute_test.cpp]: Remove 

>  Case 1.

This is very confusing for notation in tests.

File: test/module/irohad/model/command_validate_execute_test.cpp [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+    create_asset->domain_id = domain_id;
+    create_asset->precision = 2;
+
+    command = create_asset;
+  }
+
+  std::shared_ptr<CreateAsset> create_asset;
+};
+
+TEST_F(CreateAssetTest, ValidWhenCreatorHasPermissions) {
```
Comment (muratovv) [inline test/module/irohad/model/command_validate_execute_test.cpp]: Add description to test

File: test/module/irohad/model/command_validate_execute_test.cpp [hunk 6]
```diff
@@ -0,0 +1,756 @@
...
+    remove_signatory->account_id = account_id;
+    remove_signatory->pubkey = creator.master_key;
+
+    command = remove_signatory;
+  }
+
+  std::shared_ptr<RemoveSignatory> remove_signatory;
+};
+
+TEST_F(RemoveSignatoryTest, ValidWhenCreatorHasPermissions) {
```
Comment (muratovv) [inline test/module/irohad/model/command_validate_execute_test.cpp]: Add test description

**Pull request merged**